core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -186,4 +194,34 @@ public Transaction newTransaction() {
   public String toString() {
     return name();
   }
+
+  abstract Object writeReplace();
+
+  static class TableStub implements Serializable {
```

Comment:
For [other classes](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/expressions/SerializationProxies.java), we call these classes `SomethingProxy`. Should we use the same term, `Proxy`, instead of `Stub`?

Reply:
Renamed here, and in the BaseTable too.